firestore.rules

```diff
@@ -135,33 +129,138 @@ service cloud.firestore {
     match /authed/{portal} {
       allow read, write: if isAuthedTeacher();
 
+      function getTeacherNetworks() {
+        let platformUserId = request.auth.token.platform_user_id;
+        return get(/databases/$(database)/documents/authed/$(portal)/users/$(platformUserId)).data.networks;
+      }
+
+      function resourceInTeacherNetworks() {
+        return resource.data.network in getTeacherNetworks();
+      }
+
+      match /users/{userId} {
+        // teachers can read their own user documents
+        allow read: if isAuthedTeacher() && (request.auth.token.platform_user_id == userId);
+        // currently, only admins can write user information
+        allow write: if false;
+      }
+
+      match /curriculum/{pathId} {
+        // portal-authenticated teachers can create valid curriculum documents
+        allow create: if isAuthedTeacher() && isValidCurriculumCreateRequest();
+        // curriculum documents can't be updated or deleted
+        allow update, delete: if false;
+        // teachers can read their own curriculum documents or others in their network
+        allow read: if isAuthedTeacher() && (resourceMatchUserId() || resourceInTeacherNetworks());
+
+        function getCurriculumOwner() {
+          return get(/databases/$(database)/documents/authed/$(portal)/curriculum/$(pathId)).data.uid;
```

Comment:
This seemed strange when I first saw it. Then I realized it was basically looking up the parent document so its field value could be used in rules for child documents. A comment here and in getCurriculumNetwork would be useful.

It would might be more efficient to put the `uid` and `network` in the comment documents so these lookup are not necessary, but that would also mean the `uid` and `network` would have to be included in the query so the rules could verify all returned documents would be valid without actually checking them all. Because of this complexity the current approach seems good.

Reply:
So, `uid` and `network` are in the individual comment documents as well. The `uid` fields will differ, however, because different users can comment on the same networked document. The point of looking up the curriculum/document owner is that if you own the document, you can see all the comments, regardless of who wrote them. But the concept of ownership for a curriculum document is somewhat ambiguous anyway. It probably doesn't matter in the end since the network rule may be sufficient in the end.